lib/spack/spack/cmd/__init__.py

```diff
@@ -181,6 +181,23 @@ def parse_specs(args, **kwargs):
         raise spack.error.SpackError(msg)
 
 
+def matching_spec_from_env(spec):
+    """
+    Returns a concrete spec, matching what is available in the environment.
+    If no matching spec is found in the environment, this will return the
+    given spec but concretized.
+    """
+    env = spack.environment.get_env({}, cmd_name)
+    spec_from_env = None
+    if env:
+        spec_from_env = env.matching_spec(spec)
+    if spec_from_env:
+        spec = spec_from_env
+    else:
+        spec.concretize()
```

Comment:
I think this can be a little simpler
```suggestion
    if env:
        return env.matching_spec(spec)
    else:
        return spec.concretized()
```

Reply:
I don't think that diff covers the case that `env.matching_spec` returns `None`. As it is, if the user-provided spec doesn't match anything in the environment, then it reverts to the behavior before this PR: concretize separately.

Reply:
That line can be `return env.matching_spec(spec) or spec.concretized()`